packages/react-ui-core/src/DropDown/DropDown.js

```diff
@@ -0,0 +1,96 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import themed from 'react-themed'
+import classnames from 'classnames'
+import DefaultAnchor from './DefaultAnchor'
+
+@themed(/^DropDown/)
+
+export default class DropDown extends Component {
+  static propTypes = {
+    className: PropTypes.string,
+    visible: PropTypes.bool,
+    theme: PropTypes.object,
+    children: PropTypes.node,
+    Anchor: PropTypes.oneOfType([
+      PropTypes.node,
+      PropTypes.func,
+    ]),
+    toggleVisibilty: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    visible: false,
+    theme: {},
+    Anchor: DefaultAnchor,
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = { visible: this.props.visible }
+    this.handleDocumentClick = this.handleDocumentClick.bind(this)
+    this.toggleVisibilty = this.toggleVisibilty.bind(this)
+  }
+
+  componentDidMount() {
+    document.addEventListener('click', this.hanldeDocumentClick)
+  }
+
+  componentWillUnmount() {
+    document.removeEventListener('click', this.hanldeDocumentClick)
+  }
+
+  get visibleTheme() {
+    const { theme } = this.props
+    return this.state.visible ? 'visible' : 'invisible'
+  }
+
+  toggleVisibilty() {
+    this.setState({ visible: !this.state.visible })
+  }
+
+  handleDocumentClick(event) {
+    if (this.state.visible && !this.ref.contains(event.target)) {
+      this.toggleVisibilty()
+    }
+  }
+
+  render() {
+    const {
+      theme,
+      visible,
+      children,
+      Anchor,
+      className,
+      ...props
+    } = this.props
+
+    const dropDownVisible = this.state.visible
+
+    return (
+      <div
+        ref={ref => { this.ref = ref }}
+        className={classnames(
+          theme.DropDown,
+          className
+        )}
+      >
+        <Anchor
+          {...props}
+          handleDocumentClick={this.handleDocumentClick}
+          toggleVisibilty={this.toggleVisibilty}
+          dropDownVisible={dropDownVisible}
+          className={classnames(
+            theme.DropDown_Anchor,
+            theme[`DropDown_Anchor_${this.visibleTheme}`],
```

Comment:
`visibleTheme` is probably a bad name. It's not a "theme" per se, it's just a modifier for a class.  Can't you just do something like this without needing the extra function:

```
theme[`DropDown_Anchor-${dropDownVisible ? 'visible' : 'hidden'}`]
```
Also, since it is a modifier, you should be using `-` instead of `_`.

Reply:
👍